### DISCUSSION TITLE
Tab completion

### DISCUSSION THREAD

**Discussion by lUNuXl** [fish-shell/fish-shell#8751] [Q&A] [answered]
Hello,

I've set fish to complete words on pressing tab which is more in-line with basically every other software I'm using.
However I noticed that this behaviour makes it impossible for me to cd into directories that I wrongly guessed the name of (for example wrong capitalisation) in the past, example:

```
larry@toaster ~> cd git/youtubeDlDocker
cd: The directory 'git/youtubeDlDocker' does not exist
larry@toaster ~ [1]> cd git/YoutubeDlDocker
cd: The directory 'git/YoutubeDlDocker' does not exist
larry@toaster ~ [1]> cd git
larry@toaster ~/git> cd youtubeDlDocker/
cd: The directory 'youtubeDlDocker/' does not exist
```

As you can see/guess when I'm typing `youtube` and press tab I expect fish to either complete the name & fix capitalisation or give me the `fancy list` of directories that fuzzy match the name I typed in. What happens instead is that fish sees that I typed in a very simmilar command in the past, so for sure I want to cd into directory that doesn't exist.

Is there any way to fix this?

**Comment by faho** [comment, marked as answer]
The autosuggestions, that greyed-out part of the commandline, very much *aren't* guaranteed to be correct or even to exist.

They're taken from history and are the best guess we have at the moment. We have no idea if your history was wrong, or if it will start being correct, so we can simply guess.

By triggering them on tab you have removed actual, proper, logical tab completion from your shell.

>Is there any way to fix this?

The way to "fix" it is to not treat autosuggestions as a thing they are not. I.e. don't "set fish to complete words on pressing tab". It's not doing what you think it does.
